src/Sdk/WebApi/WebApi/ResultsHttpClient.cs

```diff
@@ -214,9 +214,23 @@ private async Task UploadBlockFileAsync(string url, string blobStorageType, File
             if (m_useSdk && blobStorageType == BlobStorageTypes.AzureBlobStorage)
             {
                 var blobClient = GetBlobClient(url);
+                var httpHeaders = new BlobHttpHeaders();
+                var conditions = new BlobRequestConditions
+                {
+                    IfMatch = new ETag("*")
+                };
+                string extension = System.IO.Path.GetExtension(file.Name);
+                if (extension == ".txt")
```

Comment:
is this check required, since we don't have this in the non-SDK version?

Reply:
This is needed to support https://github.com/actions/runner/pull/3114 since those should be of type `application/octet-stream`. I didn't add it to the non-sdk version since that should be rolled out by the time the diagnostic change is out, but happy to add to both if I misunderstood! CC @yacaovsnc

Reply:
I am not quite sure whether we use `.txt` in the runner, ex: are those log file ends up with `.txt`?

Reply:
Yep! Our log files our `.txt` (that's the `job-logs.txt`): https://github-grid.enterprise.slack.com/files/U02V5TYNARX/F06GB94RRK4/-.txt

Also have validated the change that it does set the content-type correctly.

The diagnostic logs are `.zip`

Reply:
Chatted offline and updated to not use the runner file extension, but the blob url instead!